File: documentation/user/en/query/ordering/examples/reference/reference-attribute-natural-hierarchy.cs
EvitaResponse<SealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("code", "accessories")
        		)
        	),
        	OrderBy(
        		ReferenceProperty(
        			"categories",
        			AttributeNatural("orderInCategory", Asc)
        		)
        	),
        	Require(
        		EntityFetch(
        			AttributeContent("code"),
        			ReferenceContentWithAttributes(
        				"categories",
        				AttributeContent("orderInCategory")
        			)
        		)
        	)
        )
	)
);

File: documentation/user/en/query/ordering/examples/reference/reference-attribute-natural-hierarchy.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					hierarchyWithin(
						"categories",
						attributeEquals("code", "accessories")
					)
				),
				orderBy(
					referenceProperty(
						"categories",
						traverseByEntityProperty(
							BREADTH_FIRST,
							entityPrimaryKeyNatural(ASC)
						),
						attributeNatural("orderInCategory", ASC)
					)
				),
				require(
					entityFetch(
						attributeContent("code"),
						referenceContentWithAttributes(
							"categories",
							attributeContent("orderInCategory")
						)
					)
				)
			)
		);
	}
);